client/scss/layouts/_page-editor.scss

```diff
@@ -622,6 +630,204 @@ footer .preview {
+    }
+
+    .preview-panel__iframe {
+      width: 768px;
+      height: calc(100% / min(1, calc(446 / 768)));
+      zoom: min(1, calc(446 / 768));
+      transform: scale(min(1, calc(446 / 768)));
+    }
+  }
+
+  // Desktop 1280x720
+  &--desktop {
+    .preview-panel__wrapper {
+      width: calc(1280px * min(1, calc(446 / 1280)));
+    }
+
+    .preview-panel__iframe {
+      width: 1280px;
+      height: calc(100% / min(1, calc(446 / 1280)));
+      zoom: min(1, calc(446 / 1280));
+      transform: scale(min(1, calc(446 / 1280)));
+    }
+  }
+
+  // Required for Safari and Chromium-based browsers
+  &--mobile,
+  &--tablet,
+  &--desktop {
+    @media screen and (-webkit-min-device-pixel-ratio: 0) {
+      .preview-panel__iframe {
+        zoom: 1;
+      }
+    }
+  }
+
+  &__area {
+    width: 100%; // 446px
+    height: 100%;
+    overflow: scroll;
+    display: flex;
+    flex-direction: column;
+    justify-content: space-between;
+  }
+
+  &__sizes {
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    column-gap: 0.75rem;
+    border-bottom: 1px solid $color-grey-4;
+    padding-bottom: 1rem;
+    margin-bottom: 1rem;
+  }
+
+  &__size-button {
+    @apply w-text-grey-400 w-transition hover:w-transform hover:w-scale-110 hover:w-text-primary focus:w-text-primary;
+
+    width: 2rem;
+    height: 2rem;
+    background: transparent;
+    box-sizing: border-box;
+    padding: 0;
+    border-radius: 5px;
+
+    .icon {
+      width: 1rem;
+      height: 1rem;
+    }
+
+    &--new-tab {
+      .icon {
+        width: 0.9rem;
+        height: 0.9rem;
+      }
+    }
+
+    &--tablet,
+    &--desktop {
+      .icon {
+        width: 1.25rem;
+        height: 1.25rem;
+      }
+    }
+  }
+
+  &__refresh-button.button--icon {
+    display: flex;
+    align-items: center;
+    gap: 0.5rem;
+    position: absolute;
+    top: 1.25rem;
+    right: 1.5rem;
+
+    .icon {
+      @include svg-icon(0.9rem);
+    }
+  }
+
+  &--mobile &__size-button--mobile,
+  &--tablet &__size-button--tablet,
+  &--desktop &__size-button--desktop {
+    @apply w-bg-primary w-text-white w-transform-none;
+  }
+
+  &__modes {
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    gap: 1rem;
+    border-top: 1px solid $color-grey-4;
+    padding-top: 1rem;
+    margin-top: 1rem;
+    margin-bottom: 0.5rem;
+  }
+
+  &__mode-select {
+    font-size: 1em;
+    padding: 0.6rem 0.75rem;
+    padding-inline-end: 2.5rem;
+    width: auto;
+    background-color: $color-white;
+  }
+
+  &__modes.choice_field &__mode-select ~ span:after {
+    @apply w-text-grey-400;
+    border: none;
+    font-size: 1.5em;
+    line-height: 1.65em;
+    top: auto;
+    width: 2.5rem;
+  }
+}
+
+.preview-error {
+  position: absolute;
+  left: 0;
+  width: 100%;
+  min-height: 100vh;
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+  text-align: center;
+  background: $color-white;
```

Comment:
The absolute positioning is needed because even the most basic template (`skeleton.html`) already has some padding applied to compensate the sidebar:

<img width="867" alt="image" src="https://user-images.githubusercontent.com/6379424/176233632-26374939-4427-4cb9-8266-2ca1ffcf67b4.png">

Reply:
Cool! I somehow missed this.